readme.md
## [![npm][npmjs-img]][npmjs-url] [![mit license][license-img]][license-url] [![build status][travis-img]][travis-url] [![coverage status][coveralls-img]][coveralls-url] [![deps status][daviddm-img]][daviddm-url]

> Get day number of the current date/year or given valid `Date` string format

## Install
```
npm i --save current-day-number
npm test
current-day-number --help
```


## API
> For more use-cases see the [tests](./test.js)

### [currentDayNumber](./index.js#L42)
> Get day number of the current date/year or given valid `Date` string format

- `[date]` **{String}** every valid Date-ish string format
- `return` **{Number}**

**Example:**

```js
var currentDayNumber = require('current-day-number');

// june 27, 2014
currentDayNumber();
//=> 177

currentDayNumber('');
//=> 177

currentDayNumber('August 07, 2015');
//=> 218

currentDayNumber('August 07, 2016');
//=> 219, cuz' year is leap

currentDayNumber('02/16/2015');
//=> 47
```

## CLI
> You can just run `current-day-number --help` for more information.

![current day number cli image](http://i.gyazo.com/f8e629c80cb2ea5641f251f3c017fecb.png)


## Author
**Charlike Mike Reagent**
+ [gratipay/tunnckoCore][author-gratipay]
+ [twitter/tunnckoCore][author-twitter]
+ [github/tunnckoCore][author-github]
+ [npmjs/tunnckoCore][author-npmjs]
+ [more ...][contrib-more]


## License [![MIT license][license-img]][license-url]
Copyright (c) 2014-2015 [Charlike Mike Reagent][contrib-more], [contributors][contrib-graf].  
Released under the [`MIT`][license-url] license.


[npmjs-url]: http://npm.im/current-day-number
[npmjs-img]: https://img.shields.io/npm/v/current-day-number.svg?style=flat&label=current-day-number

[coveralls-url]: https://coveralls.io/r/datetime/current-day-number?branch=master
[coveralls-img]: https://img.shields.io/coveralls/datetime/current-day-number.svg?style=flat

[license-url]: https://github.com/datetime/current-day-number/blob/master/license.md
[license-img]: https://img.shields.io/badge/license-MIT-blue.svg?style=flat

[travis-url]: https://travis-ci.org/datetime/current-day-number
[travis-img]: https://img.shields.io/travis/datetime/current-day-number.svg?style=flat

[daviddm-url]: https://david-dm.org/datetime/current-day-number
[daviddm-img]: https://img.shields.io/david/datetime/current-day-number.svg?style=flat

[author-gratipay]: https://gratipay.com/tunnckoCore
[author-twitter]: https://twitter.com/tunnckoCore
[author-github]: https://github.com/tunnckoCore
[author-npmjs]: https://npmjs.org/~tunnckocore

[contrib-more]: http://j.mp/1stW47C
[contrib-graf]: https://github.com/datetime/current-day-number/graphs/contributors

***

_Powered and automated by [kdf](https://github.com/tunnckoCore), March 02, 2015_

history.md


## v1.0.6 / March 02, 2015
- Release v1.0.6 / npm@v1.0.6
- add more examples to help info
- run `update readme`
- fix/faster travis
- bump deps
- add license banner
- add cli

## v1.0.5 / January 26, 2015
- Release v1.0.5 / npm@v1.0.5
- bump to day-seconds@^1.0.0

## v1.0.4 / January 26, 2015
- Release v1.0.4 / npm@v1.0.4
- update tests
- add coverage
- refactor, docs

## v1.0.3 / December 12, 2014
- Release v1.0.3 / npm@v1.0.3
- add check for empty string

## v1.0.2 / December 12, 2014
- Release v1.0.2 / npm@v1.0.2
- MAKEFILE!

## v1.0.1 / December 12, 2014
- Release v1.0.1 / npm@v1.0.1
- tweaks, ~~Makefile (disappears, wtf?)~~
- add and fix tests

## v1.0.0 / December 12, 2014
- Release v1.0.0 / npm@v1.0.0
- moved to [datetime org](https://github.com/datetime)
- fixes/tweaks readme, licenses, Makefile, package.json, dotfiles, etc

## v0.1.0 / June 27, 2014
- Release v0.1.0 / npm@v0.1.0
- init commits


test.js
/**
 * current-day-number <https://github.com/datetime/current-day-number>
 *
 * Copyright (c) 2014-2015 Charlike Mike Reagent, contributors.
 * Released under the MIT license.
 */

'use strict';

var assert = require('assert');
var currentDayNumber = require('./index');

describe('current-day-number:', function() {
  it('should be greater than 177 or 1 (june 27, 2014 - released)', function(done) {
    assert.strictEqual(typeof currentDayNumber(), 'number');
    done();
  });
  it('should be 218 for "August 07, 2015" and 47 for "02/16/2015"', function(done) {
    assert.strictEqual(currentDayNumber('March 24, 2015'), 83);
    assert.strictEqual(currentDayNumber('03/24/2016'), 84);
    assert(currentDayNumber('August 07, 2015') >= 218);
    assert.strictEqual(currentDayNumber('02/16/2015'), 47);
    assert.strictEqual(currentDayNumber('12/15/2014'), 349);
    done();
  });
  it('should get current day number when empty string format', function(done) {
    assert.strictEqual(typeof currentDayNumber(''), 'number');
    done();
  });
  it('should get current day number, if Date format string is invalid', function(done) {
    assert.strictEqual(typeof currentDayNumber(/regex/g), 'number');
    assert.strictEqual(typeof currentDayNumber(true), 'number');
    assert.strictEqual(typeof currentDayNumber(false), 'number');
    assert.strictEqual(typeof currentDayNumber(222), 'number');
    done();
  });
});


index.js
/**
 * current-day-number <https://github.com/datetime/current-day-number>
 *
 * Copyright (c) 2014-2015 Charlike Mike Reagent, contributors.
 * Released under the MIT license.
 */

'use strict';

var daySeconds = require('day-seconds');

/**
 * Get day number of the current date/year or given valid `Date` string format
 *
 * **Example:**
 *
 * ```js
 * var currentDayNumber = require('current-day-number');
 *
 * // june 27, 2014
 * currentDayNumber();
 * //=> 177
 *
 * currentDayNumber('');
 * //=> 177
 *
 * currentDayNumber('August 07, 2015');
 * //=> 218
 *
 * currentDayNumber('August 07, 2016');
 * //=> 219, cuz' year is leap
 *
 * currentDayNumber('02/16/2015');
 * //=> 47
 * ```
 *
 * @name currentDayNumber
 * @param  {String} `[date]` every valid Date-ish string format
 * @return {Number}
 * @api public
 */
module.exports = function currentDayNumber(date) {
  var instance = new Date();

  if (typeof date === 'string' && date.length) {
    instance = new Date(date);
  }

  var now = instance.getTime();
  var start = instance.setMonth(0,0);
  var diff = now - start;

  return (diff / daySeconds(false)) | 0;
};
